### DISCUSSION TITLE
Integration and Repository Isolation in Micronaut Data: Combining JDBC, JPA, and Hibernate Reactive

### DISCUSSION THREAD

**Discussion by UltimaPhoenix** [micronaut-projects/micronaut-data#2635] [Q&A] [answered]
Hello Micronaut Data team,

I'm exploring the capabilities of Micronaut Data for a project, and I'm curious about the possibility of using Micronaut Data JDBC, Micronaut Data JPA, and Micronaut Data Hibernate Reactive together within the same Micronaut application. Specifically, I'm interested in using different parts/features of Micronaut data in different parts of my application.

Could you provide guidance on how I can seamlessly integrate these components in distinct sections of my application? Additionally, I'm curious about the best practices for isolating repositories, (JDBC, JPA, etc..) to ensure modularity and maintainability. How can I structure my repositories to keep functionalities isolated?

Thank you for your assistance!

Best regards

**Comment by dstepanov** [comment, marked as answer]
You can take a look at this guide https://guides.micronaut.io/latest/micronaut-data-r2dbc-repository-gradle-java.html and just add Micronaut Data JPA.
You can use one base repository and three repositories that extend it with different annotations like `@JdbcRepositor`, `@R2dbcRepositotry` etc
